File: binary_period.py
import numpy as np 

def binary_period(number):
    zeros = [0] * 30
    idx = 0 
    while number > 0:
        zeros[idx] = number % 2 
        number //= 2
        idx += 1 
    for c in range(1, 1+idx):
        check = True
        for i in range(idx-c):
            if zeros[i] != zeros[i+c]:
                check = False 
                break 
            
        if check:
            return c 
            
    return -1

    
if __name__ == '__main__':
    print(binary_period(955))

File: binary_period.cs
namespace Namespace {
    
    using np = numpy;
    
    using System.Collections.Generic;
    
    using System;
    
    using System.Linq;
    
    public static class Module {
        
        public static object binary_period(object number) {
            var zeros = new List<object> {
                0
            } * 30;
            var idx = 0;
            while (number > 0) {
                zeros[idx] = number % 2;
                number /= 2;
                idx += 1;
            }
            foreach (var c in Enumerable.Range(1, 1 + idx - 1)) {
                var check = true;
                foreach (var i in Enumerable.Range(0, idx - c)) {
                    if (zeros[i] != zeros[i + c]) {
                        check = false;
                        break;
                    }
                }
                if (check) {
                    return c;
                }
            }
            return -1;
        }
    }
}
